src/main/java/org/myrobotlab/service/meta/abstracts/MetaData.java

```diff
@@ -0,0 +1,418 @@
+package org.myrobotlab.service.meta.abstracts;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+
+import org.myrobotlab.codec.CodecUtils;
+import org.myrobotlab.framework.ServiceReservation;
+import org.myrobotlab.framework.repo.ServiceArtifact;
+import org.myrobotlab.framework.repo.ServiceData;
+import org.myrobotlab.framework.repo.ServiceDependency;
+import org.myrobotlab.framework.repo.ServiceExclude;
+import org.myrobotlab.logging.LoggerFactory;
+import org.slf4j.Logger;
+
+/**
+ * MetaData describes most of the data about a service that is static.
+ * It's dependencies, categories and other meta information.
+ * 
+ * It also describes its list of peers. They define a list of group this 
+ * service expects to interact with.  When a service is started it gets
+ * a copy of MetaData based on its instance name. When a new instance is
+ * created a list of ServiceData.overrides are consulted, to allow the user
+ * to override actual name and type information.
+ * 
+ */
+public class MetaData implements Serializable {

```

Comment:
This class should me marked as abstract and probably renamed to AbstractMetaData.

Reply:
done.